### PR TITLE
fix(FEC-10732, FEC-10759): player params are not injected to additional instances' config

### PR DESCRIPTION
**Pull request by yairans**
### Description of the Changes

Issue: `plugins-config-store.js` [deletes](https://github.com/kaltura/kaltura-player-js/blob/master/src/common/plugins/plugins-config-store.js#L110) the resolved params from the static plugins config so they are missing for the following instances. 
Solution: Create an instance of plugins config for each player

Solves FEC-10732, FEC-10759

### CheckLists

- [x] changes have been done against master branch, and PR does not conflict
- [x] new unit / functional tests have been added (whenever applicable)
- [x] test are passing in local environment
- [x] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

**Review by dan-ziv — DISMISSED** [review]

File: src/common/plugins/plugins-config-store.js [hunk 1]
```diff
@@ -88,36 +87,43 @@ const removeUndefineds = (obj: Object = {}): Object =>
     return product;
   }, {});
 
-const pluginConfig = {
+class pluginConfigStore {
```
Comment (dan-ziv) [inline src/common/plugins/plugins-config-store.js:90]: If it's a class now, change to CamelCase

File: src/common/plugins/plugins-config.js [hunk 2]
```diff
@@ -190,4 +151,52 @@ function _mergeConfig(data: Object, evaluatedConfig: Object): void {
   }
 }
 
-export {evaluatePluginsConfig, evaluateUIConfig, getEncodedReferrer};
+class PluginsConfigure {
```
Comment (dan-ziv) [inline src/common/plugins/plugins-config.js:154]: `class ConfigEvaluator`

File: src/kaltura-player.js [hunk 3]
```diff
@@ -52,12 +52,14 @@ class KalturaPlayer extends FakeEventTarget {
   _firstPlay: boolean = true;
   _sourceSelected: boolean = false;
   _pluginReadinessMiddleware: PluginReadinessMiddleware;
+  _pluginsConfigure: PluginsConfigure;
 
-  constructor(options: KPOptionsObject) {
+  constructor(options: KPOptionsObject, pluginsConfigure: PluginsConfigure) {
```
Comment (dan-ziv) [inline src/kaltura-player.js:57]: Why as an argument and not as class member that you are initialize internally?

File: src/common/plugins/plugins-config-store.js (hunk 1 above)
Comment (yairans) [reply src/common/plugins/plugins-config-store.js]: done

File: src/kaltura-player.js (hunk 3 above)
Comment (yairans) [reply src/kaltura-player.js]: we already initialize the PluginsConfigure in `setup`

Comment (dan-ziv) [reply src/kaltura-player.js]: so don't, initialize it here

Comment (yairans) [reply src/kaltura-player.js]: I can't. see https://github.com/kaltura/kaltura-player-js/blob/FEC-10732/src/setup.js#L33

Comment (dan-ziv) [reply src/kaltura-player.js]: Again, you can do it inside the constructor.

Comment (yairans) [reply src/kaltura-player.js]: you are right. done.

File: src/common/plugins/plugins-config.js (hunk 2 above)
Comment (yairans) [reply src/common/plugins/plugins-config.js]: done

**Review by dan-ziv — DISMISSED** [review]

**Review by dan-ziv — APPROVED** [review]

**Comment by dan-ziv** [comment]
Also solves https://kaltura.atlassian.net/browse/FEC-10759

@yairans I've change the title since it's not specific issue related to kava

**Review by Yuvalke — APPROVED** [review]

**Review by OrenMe — APPROVED** [review]

**Pull request merged**